datafusion/src/optimizer/constant_folding.rs

```diff
@@ -106,16 +106,23 @@ impl OptimizerRule for ConstantFolding {
                             .rewrite(&mut const_evaluator)?
                             .rewrite(&mut simplifier)?;
 
-                        let new_name = &new_e.name(plan.schema());
-
-                        if let (Ok(expr_name), Ok(new_expr_name)) = (name, new_name) {
-                            if expr_name != new_expr_name {
-                                Ok(new_e.alias(expr_name))
-                            } else {
-                                Ok(new_e)
+                        match plan {
+                            LogicalPlan::Filter { .. } => Ok(new_e),
```

Comment:
I would like to recommend adding a comment here about why we don't add aliases for filter.